apps/src/lib/levelbuilder/lesson-editor/VocabulariesEditor.jsx

```diff
@@ -0,0 +1,161 @@
+  editVocabulary,
+  removeVocabulary
+} from '@cdo/apps/lib/levelbuilder/lesson-editor/vocabulariesEditorRedux';
+
+const styles = {
+  oddRow: {
+    backgroundColor: color.lightest_gray
+  },
+  actionsColumn: {
+    display: 'flex',
+    justifyContent: 'space-evenly',
+    backgroundColor: 'white'
+  },
+
+  remove: {
+    fontSize: 14,
+    color: 'white',
+    background: color.dark_red,
+    cursor: 'pointer',
+    textAlign: 'center',
+    width: '98%'
+  }
+};
+
+class VocabulariesEditor extends Component {
+  static propTypes = {
+    courseVersionId: PropTypes.number,
+
+    // Provided by redux
+    vocabularies: PropTypes.arrayOf(vocabularyShape).isRequired,
+    addVocabulary: PropTypes.func.isRequired,
+    editVocabulary: PropTypes.func.isRequired,
+    removeVocabulary: PropTypes.func.isRequired
+  };
+
+  constructVocabularyOption = vocabulary => ({
+    value: vocabulary.key,
+    label: `${vocabulary.word} - ${vocabulary.definition}`,
+    vocabulary
+  });
+
+  constructor(props) {
+    super(props);
+    this.state = {confirmRemovalDialogOpen: false, vocabularyForRemoval: null};
+  }
+
+  constructSearchOptions = json => {
+    const vocabKeysAdded = this.props.vocabularies.map(vocab => vocab.key);
+    const vocabularies = json
+      .map(vocab => this.constructVocabularyOption(vocab))
+      .filter(vocab => vocabKeysAdded.indexOf(vocab.value) === -1);
+    return {options: vocabularies};
+  };
+
+  handleRemoveVocabularyDialogClose = () => {
+    this.setState({
+      confirmRemovalDialogOpen: false,
+      vocabularyForRemoval: null
+    });
+  };
+
+  handleRemoveVocabularyConfirm = () => {
+    this.props.removeVocabulary(this.state.vocabularyForRemoval.key);
+    this.handleRemoveVocabularyDialogClose();
+  };
+
+  render() {
+    return (
+      <div>
+        {this.state.confirmRemovalDialogOpen && (
+          <Dialog
+            body={`Are you sure you want to remove the vocabulary "${
+              this.state.vocabularyForRemoval.word
+            }" from this lesson?`}
+            cancelText="Cancel"
+            confirmText="Delete"
+            confirmType="danger"
+            isOpen={this.state.confirmRemovalDialogOpen}
+            handleClose={this.handleRemoveVocabularyDialogClose}
+            onCancel={this.handleRemoveVocabularyDialogClose}
+            onConfirm={this.handleRemoveVocabularyConfirm}
+          />
+        )}
+        <input
+          type="hidden"
+          name="vocabularies"
+          value={JSON.stringify(this.props.vocabularies.map(v => v.key))}
+        />
+        <div>
+          Select a vocabulary word to add
+          <SearchBox
+            onSearchSelect={e => this.props.addVocabulary(e.vocabulary)}
+            additionalQueryParams={{
+              courseVersionId: this.props.courseVersionId
+            }}
+            searchUrl={'vocabularysearch'}
+            constructOptions={this.constructSearchOptions}
+          />
+        </div>
+        <div>
```

Comment:
Since the tables are very similar for each of these editors would it make sense to create a component that could be re-used and contain the styling for the table?

Reply:
Oh interesting. I thought about trying to combine the resources and vocabulary components but never though to just pull out the table. I like that -- I'll play around and see how hard it would be.

Reply:
I'm not sure if it is too late for this, but I know we also use a table component many other places in our react code. I believe it is reactabular-table . it could be worth looking around to see if you could just use that component in both places.

Reply:
Briefly looking at the documentation for `reactabular-table` it looks like it might be difficult to add buttons for editing and removing. We could change how that looks of course but I'm not sure the best way of doing that.

Reply:
It could be worth looking at the ManageStudentsTable, which uses buttons for editing and removing. No pressure to adopt this now, but wanted to list it as an option if what you're looking to extract is abstract table functionality.

Reply:
Yeah, I'm tempted to merge this then look at trying to unify these tables. Whatever we decide on will change ResourceEditor and maybe ObjectivesEditor anyway.